### PR TITLE
Batchnorm transform

### PR DESCRIPTION
**Pull request by stefanwebb**
Implementation of batch-normalization transform, used to stabilize learning in normalizing flows. As requested in #1517

### CODE REVIEW

File: pyro/distributions/batch_norm.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    running averages of the sample mean and variance, respectively. The constraint :math:`\\gamma>0` is enforced to
+    ease calculation of the log-det-Jacobian term.
+
+    This is an element-wise transform, and when applied to a vector, learns two parameters (:math:`\\gamma,\\beta`)
+    for each dimension of the input.
+
+    When the module is set to training mode, the moving averages of the sample mean and variance are updated every time
+    the inverse operator is called, e.g., when a normalizing flow scores a minibatch with the `log_prob` method.
+
+    Also, when module is set to training mode, the sample mean and variance on the current minibatch are used in place
```
Comment (martinjankowiak) [inline pyro/distributions/batch_norm.py]: 'the module'?

Comment (stefanwebb) [reply pyro/distributions/batch_norm.py]: Fixed
